tools/base/fpgaconf/bitstream-tools.h

```diff
@@ -168,61 +151,45 @@ fpga_result check_bitstream_guid(const uint8_t *bitstream)
 	return FPGA_OK;
 }
 
-fpga_result get_fpga_interface_id(fpga_token token, uint64_t *id_l,
-								uint64_t *id_h)
+
+fpga_result get_fpga_interface_id(fpga_token token, fpga_guid interface_id)
 {
 	fpga_result result = FPGA_OK;
-	char buf_l[INTFC_ID_LOW_LEN + 1] = { 0 };
-	char buf_h[INTFC_ID_HIGH_LEN + 1] = { 0 };
-	uint8_t buffer[INTFC_ID_LOW_LEN + INTFC_ID_HIGH_LEN] = { 0 };
-
-	fpga_object pr_object;
+	fpga_result resval = FPGA_OK;
+	fpga_properties filter = NULL;
+	fpga_guid guid;
 	errno_t e;
 
-	result = fpgaTokenGetObject(token, PR_INTERFACE_ID, &pr_object, 0);
+	result = fpgaGetProperties(token, &filter);
 	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to get Token Object \n");
-		return result;
+		OPAE_ERR("Failed to get Token Properties Object \n");
+		goto out;
 	}
-
-	result = fpgaObjectRead(pr_object, buffer, 0,
-		INTFC_ID_LOW_LEN + INTFC_ID_HIGH_LEN, 0);
+	result = fpgaPropertiesGetGUID(filter, &guid);
```

Comment:
Do we want to add a safeguard so that this function fails if the device type is not FPGA_DEVICE?

Reply:
fixed